tests/nlu/parsers/test_integer.py

```diff
@@ -27,7 +23,85 @@ def test_integer():
     parsed = integer.parse(metadata, raw_value)
     assert parsed == 404
 
-    metadata = {"range": [1, 500]}
+
+def test_exponent():
+    metadata = {"range": [1, 10000]}
+    raw_value = "nine thousand one"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 9001
+
+    raw_value = "one hundred"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 100
+
+
+def test_teens():
+    metadata = {"range": [1, 19]}
+    raw_value = "thirteen"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 13
+
+    raw_value = "13"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 13
+
+    raw_value = "fourteen"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 14
+
+    raw_value = "fifteen"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 15
+
+    raw_value = "sixteen"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 16
+
+
+def test_multiples_of_ten():
+    metadata = {"range": [1, 100]}
+    raw_value = "ten"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 10
+
+    raw_value = "10"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 10
+
+    raw_value = "20"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 20
+
+    raw_value = "30"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 30
+
+    raw_value = "44"
+    parsed = integer.parse(metadata, raw_value)
+    assert parsed == 44
+
+
+def test_out_of_range():
```

Comment:
I think a test over ordinals would be useful too. The [inflect](https://pypi.org/project/inflect/) library could be useful for generating values for these tests; you could just loop over every number or a sample of numbers from `1..n` and run it through the parser in its numeric, English cardinal, and English ordinal form, comparing the result to the input.